packages/compiler/src/transpile.ts

```diff
@@ -1,20 +1,354 @@
+              ),
+              ...(componentProps ? componentProps.properties : []),
+            ];
+          }
+        }
+      },
+      ExportDeclaration(path: {
+        node: ExportDefaultDeclaration | ExportNamedDeclaration;
+        remove(): void;
+        replaceWith(
+          declaration: FunctionDeclaration | VariableDeclaration
+        ): void;
+        replaceWithMultiple(expressions: Expression[]): void;
+      }) {
+        if (t.isExportDefaultDeclaration(path.node)) {
+          let component: Identifier | undefined;
+          const declaration = (path.node as ExportDefaultDeclaration)
+            .declaration as TSAsExpression | FunctionDeclaration | Identifier;
+
+          if (t.isTSAsExpression(declaration)) {
+            // export default X;
+            component = (declaration as TSAsExpression)
+              .expression as Identifier;
+
+            path.remove();
+          } else if (t.isFunctionDeclaration(declaration)) {
+            // export default function X()
+            component = (declaration as FunctionDeclaration).id as Identifier;
+            if (!component) {
+              // export default function ()
+              component = t.identifier('BWEPlaceholderComponent');
+              (declaration as FunctionDeclaration).id = component as Identifier;
+            }
+
+            path.replaceWith(declaration as FunctionDeclaration);
+          } else if (t.isIdentifier(declaration)) {
+            component = declaration as Identifier;
+            path.remove();
+          } else {
+            console.error(`unsupported declaration type ${declaration?.type}`);
+          }
+
+          exports.default = component!.name;
+        } else if (t.isExportNamedDeclaration(path.node)) {
+          const { declaration } = path.node as ExportNamedDeclaration;
+          if (t.isVariableDeclaration(declaration)) {
+            const [exported] = (declaration as VariableDeclaration)
+              .declarations;
+            const exportedName = (exported.id as Identifier).name;
+            exports.named.push(exportedName);
+            if (exportedName === 'BWEComponent' && !exports.default) {
+              exports.default = exportedName;
+              console.warn(
+                `Component ${componentPath} relies on a named export for "BWEComponent". In future versions, the module Component must be a default export.`
+              );
+            }
+
+            path.replaceWith(declaration as VariableDeclaration);
+          } else {
+            console.error(`unsupported export type ${path.node.type}`);
+          }
+        }
+      },
+      ImportDeclaration(path: { node: ImportDeclaration; remove(): void }) {
+        const {
+          node: { source, specifiers },
+        } = path;
+
+        const importExpressions = specifiers.map((specifier) => {
+          if (t.isImportSpecifier(specifier)) {
+            // TODO differentiate/handle namespaced & side-effect imports
```

Comment:
Might be worth creating a backlog task for that // TODO.

Reply:
Pushed an update 🙂